pkg/component/gardener/dashboard/dashboard.go

```diff
@@ -138,12 +136,18 @@ func (g *gardenerDashboard) Deploy(ctx context.Context) error {
 		return err
 	}
 
+	var sessionSecretPreviousName string
+	sessionSecretPrevious, previousExists := g.getPreviousSessionSecret()
+	if previousExists {
```

Comment:
```suggestion
	if sessionSecretPrevious, ok := g.getPreviousSessionSecret(); ok {
```

Reply:
But why we need `sessionSecretPreviousName` at all? Isn't `sessionSecretPrevious` enough to pass to the `g.deployment()` call? Currently, you are checking there if the passed string (name) is non-empty. Instead, you could also simply pass the entire secret and check if it is non-nil.

Reply:
The reason I did this was because the deploy function parameters were all names and I wanted to align to that. However it indeed makes the code easier when I pass `sessionSecretPrevious` to `g.deployment()` so I changed it.